types.ts

```diff
@@ -413,6 +418,74 @@ export interface WebBasicAuthentication extends BaseAuthentication {
   };
 }
 
+/**
+ * Parameters for the {@link CustomAuthentication} authentication type.
+ */
+export interface CustomAuthParameter {
+  /**
+   * The name used to refer to this parameter and to generate the template replacement string.
+   */
+  name: string;
+
+  /**
+   * A description shown to the user indicating what value they should provide for this parameter.
+   */
+  description: string;
+}
+
+/**
+ * Authenticate for custom, non-standard API authentication schemes by inserting one or more arbitrary secret values
+ * into the request (the body, URL, headers, or form data) using template replacement.
+ *
+ * Some APIs use non-standard authentication schemes which often require secret credentials to be put in specific places
+ * in the request URL or request body. Custom authentication supports many of these cases by allowing you as the pack
+ * author to define one or more secret values that the user or you as the pack author must provide (depending on
+ * user or system authentication). When constructing a network request, you may indicate where these values should
+ * be inserted by our fetcher service using the syntax described below (similar to templating engines).
+ *
+ * To insert the credentials, simply put `{{<paramName>-<invocationToken>}}` as a string anywhere in your request,
+ * where `<paramName>` is the name of the parameter defined in the params mapping and `<invocationToken>` is the
+ * secret invocation-specific token provided within the {@link ExecutionContext}. The invocation
+ * token is required for security reasons.
+ *
+ * @example
+ * // Suppose you're using an API that requires a secret id in the request URL,
```

Comment:
Would just verify how this renders in the generated TypeDoc. I don't know what non-code looks like below an @example tag, if it looks code you could maybe remove the `//` prefixes here, or if it doesn't render well, could put the `//` comments inside the triple backticks to make it part of the code (that was my original intention, but if it doesn't need to be part of the code, it probably doesn't need comment markers at all).

Reply:
oh good call-out -- i had verified via the vscode generated comment but will validate on typedoc